src/nvim/os/env.c

```diff
@@ -557,9 +553,6 @@ void home_replace(buf_T *buf, char_u *src, char_u *dst, int dstlen, bool one)
     dirlen = STRLEN(homedir);
 
   homedir_env_orig = homedir_env = (char_u *)os_getenv("HOME");
-  /* Empty is the same as not set. */
-  if (homedir_env != NULL && *homedir_env == NUL)
-    homedir_env = NULL;
```

Comment:
I replaced `homedir_env_orig` with `bool must_free`. It looks to me like `envlen` will have the same value as the old code when `$HOME = ""`.